src/pydata_sphinx_theme/assets/styles/components/_search.scss

```diff
@@ -86,10 +86,10 @@
   margin: 15vh auto 0; // an open modal dialog has a fixed position, so these margins bring it 15% down the viewport height and center it horizontally
   width: 90%;
   max-width: 800px;
+  max-height: 80vh; // when we display search-as-you-type results in the modal, it needs to be able to expand in height
```

Comment:
Don't know why I didn't see this the first time around, but yeah, the simple solution is to apply a max-height on the search dialog, not to toggle a height class 🙈